packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphForLoopBlock.ts

```diff
@@ -51,9 +50,9 @@ export class FlowGraphForLoopBlock extends FlowGraphWithOnDoneExecutionBlock {
         let index = context._getExecutionVariable(this, "index");
         const endIndex = context._getExecutionVariable(this, "endIndex");
         if (index < endIndex) {
-            this.index.value = index;
+            this.index.setValue(index, context);
             this.onLoop._activateSignal(context);
-            const step = context._getExecutionVariable(this, "step");
+            const step = context._getExecutionVariableWithDefault(this, "step", 1);

```

Comment:
sorry I may have missed it earlier why not having an overload instead of 2 function names?